FrontEnd/composables/useAuth.js

```diff
@@ -1,47 +1,58 @@
 // composables/useAuth.js
-// import { useAuthStore } from '@/store/auth';
-import { useUserStore } from "~/stores/useUserStore";
-import { useNuxtApp } from "#app";
-import { srvLogin, srvSignup } from "~/services/authServices";
+import { useUserStore } from "~/stores/user";
+import { useCookie } from '#app';
+
 
 export const useAuth = () => {
-  const { $api } = useNuxtApp();
-  //   const authStore = useAuthStore();
+  const { $customApi } = useNuxtApp();
+  const { $toast } = useNuxtApp();
   const userStore = useUserStore();
 
-  const handleAuthError = (error) => {
-    console.error("Authentication error: ", error.message);
-  };
 
   const login = async (credentials) => {
     try {
-        const {data} = await srvLogin($api, credentials);
-        console.log("response after fetching data : ", data);
-          // localStorage.setItem('token', data.token);
-          // localStorage.setItem('userInfo', JSON.stringify(data.user) );
-          userStore.setUserInfo(data.user);
-          userStore.setToken(data.token);
-          navigateTo("/")
+        console.log("custom api : ",$customApi)
+        const response = await $customApi.post('/auth/login', credentials);
```

Comment:
create a different layer for calling api's, defining its params so it can be reused.

Reply:
created a different layer to call api's called services and also created a util function for default api which can be reused over the api calls . please review the latest commit for the changes.